code/modules/mob/living/simple_animal/simple_animal.dm

```diff
@@ -594,8 +598,25 @@ mob/living/simple_animal/bullet_act(var/obj/item/projectile/Proj)
 						UnarmedAttack(movement_target)
 						if (get_turf(movement_target) == loc)
 							set_dir(pick(NORTH, SOUTH, EAST, WEST, NORTH, NORTH))//Face a random direction when eating, but mostly upwards
-					else if(ishuman(movement_target.loc) && Adjacent(src, get_turf(movement_target)) && prob(10))
-						beg(movement_target, movement_target.loc)
+					else if(ishuman(movement_target.loc) && Adjacent(get_turf(movement_target.loc)) && (prob(10) || nutrition / max_nutrition <= 0.25))
+						var/mob/living/carbon/human/H = movement_target.loc
+						var/obj/item/weapon/reagent_containers/food/snacks/F = null
+						if(istype(H.l_hand, /obj/item/weapon/reagent_containers/food/snacks))
+							F = H.l_hand
+
+						else if(istype(H.r_hand, /obj/item/weapon/reagent_containers/food/snacks))
+							F = H.r_hand

```

Comment:
There's no `if (!F) return` command here. So the following code might well be executed with `F == null`.